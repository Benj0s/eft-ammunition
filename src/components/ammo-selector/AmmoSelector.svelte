<script>
  import Select, { Option } from "@smui/select";
  import Icon from "@smui/select/icon/index";
  import AmmoIcon from "../../theme/icons/ammo.svg";
  import { displayedAmmo } from "../../store.js";

  export let ammunition;

  let selectedAmmo;

  displayedAmmo.update(() => {
    return ammunition[0];
  });

  // having to set the selectedAmmo using label due to: https://github.com/hperrin/svelte-material-ui/issues/107
  const unsubscribe = displayedAmmo.subscribe(a => (selectedAmmo = a.label));

  $: displayedAmmo.update(() => ammunition.find(a => a.label === selectedAmmo));
</script>

<style>
  .ammo-selector {
    display: flex;
  }
</style>

<div class="ammo-selector">
  <Select
    bind:value={selectedAmmo}
    label="Ammunition"
    variant="outlined"
    withLeadingIcon>
    <span slot="icon">
      <Icon class="material-icons">
        {@html AmmoIcon}
      </Icon>
    </span>
    {#each ammunition as ammo}
      <Option value={ammo.id} selected={ammo.label === selectedAmmo}>
        {ammo.label}
      </Option>
    {/each}
  </Select>
</div>
