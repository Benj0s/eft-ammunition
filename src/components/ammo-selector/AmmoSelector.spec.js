import AmmoSelector from './AmmoSelector.svelte'
import { render } from '@testing-library/svelte'

describe('AmmoSelector', () => {
  it('should render the component', async () => {
    const { container } = render(AmmoSelector, {
      props: {
        ammunition: [
          { id: 'test-ammo-id-1', label: 'test-ammo-label-1' },
          { id: 'test-ammo-id-2', label: 'test-ammo-label-2' },
        ],
      },
    })

    expect(container).toMatchSnapshot()
  })
})
