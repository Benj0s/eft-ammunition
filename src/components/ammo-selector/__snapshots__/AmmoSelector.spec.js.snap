// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmmoSelector should render the component 1`] = `
<body>
  <div>
    <div
      class="ammo-selector svelte-xl2f45"
    >
      <div
        class="
    mdc-select
    
    
    mdc-select--outlined
    
    mdc-select--with-leading-icon
    
  "
      >
        <span
          slot="icon"
        >
          <i
            aria-hidden="true"
            class="mdc-select__icon material-icons"
            tabindex="-1"
          >
            <svg-mock />
          </i>
        </span>
         
        <i
          class="mdc-select__dropdown-icon"
        />
         
        <select
          class="mdc-select__native-control "
          id="SMUI-select-0"
        >
           
          <option>
            test-ammo-label-1
             
          </option>
          
          <option>
            test-ammo-label-2
             
          </option>
          
          
        </select>
         
         
        <div
          class="mdc-notched-outline mdc-notched-outline--upgraded mdc-notched-outline--notched"
        >
          <div
            class="mdc-notched-outline__leading"
          />
           
          <div
            class="mdc-notched-outline__notch"
            style="width: 0px;"
          >
            <label
              class="mdc-floating-label mdc-floating-label--float-above"
              for="SMUI-select-0"
              style="transition-duration: 0s;"
            >
              Ammunition
            </label>
            
            
          </div>
           
          <div
            class="mdc-notched-outline__trailing"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
